addons/web/static/src/views/fields/relational_utils.js

```diff
@@ -167,6 +167,9 @@ export class Many2XAutocomplete extends Component {
                 if (this.props.value !== autoCompleteInput.value) {
                     autoCompleteInput.value = "";
                 }
+                if (isDiscarded && !isToMany) {
```

Comment:
what about adding a `onRecordDiscarded` instead?

Reply:
I'll try that.

Reply:
It looks so much better. I also don't need to change the API.